File: AutomatedTesting/PythonApp/HomeGrownTest.py
import Conversions

def TestConversion(testDescription, actual, expected) :
    if( actual == expected) :
        print("Test "+str(testDescription)+" passed")
    else :
        print("Test "+str(testDescription)+" failed, Expected: "+str(expected)+", was: "+str(actual))

TestConversion("Room Temperature", Conversions.ToCelcius(68), 20)

TestConversion("Boiling Point", Conversions.ToCelcius(212), 100)

TestConversion("Freezing Point", Conversions.ToCelcius(32), 0)

TestConversion("Body Temperature", Conversions.ToCelcius(98.4), 36.9)

File: AutomatedTesting/CSharpApp/HomeGrownTest.cs
using System;

namespace CSharpApp
{
    class HomeGrownTest
    {
        static void Main(string[] args)
        {
            TestConversion(1, Conversions.ToCelcius(68), 20);
            TestConversion(2, Conversions.ToCelcius(212), 100);
            TestConversion(3, Conversions.ToCelcius(32), 0);

            Console.ReadKey(); //Waits for a single key input before terminating.
        }

        static void TestConversion(int testNumber, decimal actual, decimal expected)
        {
            if (actual == expected)
            {
                Console.WriteLine("Test " + testNumber + " passed");
            }
            else
            {
                Console.WriteLine("Test " + testNumber + " failed, Expected: " + expected + ", was: " + actual);
            }
        }
    }
}
